pkg/query/columnquery.go

```diff
@@ -918,31 +935,58 @@ func sliceRecord(r arrow.Record, indices []int64) []arrow.Record {
 	return slices
 }
 
-func (q *ColumnQueryAPI) getMappingFiles(
+func getMappingFilesAndLabels(
 	ctx context.Context,
-	m *pb.MergeProfile,
-) ([]string, error) {
-	p, err := q.querier.GetProfileMetadataMappings(
-		ctx,
-		m.Query,
-		m.Start.AsTime(),
-		m.End.AsTime(),
-	)
+	q Querier,
+	query string,
+	startTime, endTime time.Time,
+) ([]string, []string, error) {
+	mappingFiles, err := q.GetProfileMetadataMappings(ctx, query, startTime, endTime)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return p, nil
+	labels, err := q.GetProfileMetadataLabels(ctx, startTime, endTime)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return mappingFiles, labels, nil
 }
 
-func (q *ColumnQueryAPI) getLabels(
-	ctx context.Context,
-	req *pb.LabelsRequest,
-) ([]string, error) {
-	l, err := q.querier.GetProfileMetadataLabels(ctx, req.Match, req.Start.AsTime(), req.End.AsTime())
-	if err != nil {
-		return nil, err
+// This is a deduplicating k-way merge.
+// The two slices that are passed in are assumed to be sorted.
+func MergeTwoSortedSlices(arr1, arr2 []string) []string {
```

Comment:
nit: no need to export this function

Reply:
ah I'd exported because the function is being tested in the `columnquery.test.go` file. is there a better way to do this?

Reply:
it's still the same package, within the same package unexported functions can be used

Reply:
gotcha!